packages/babel-traverse/src/path/replacement.js

```diff
@@ -239,6 +238,8 @@ export function replaceExpressionWithStatements(nodes: Array<Object>) {
       }
     }
 
+    this.get("callee").arrowFunctionToExpression();
```

Comment:
Again just to make the tests pass, we could also leave this out and insert real arrows if we want to.